tests/IBTK/rt0_coarsen_01.cpp

```diff
@@ -0,0 +1,194 @@
+        const int u_cc_idx = var_db->registerVariableAndContext(u_cc_var, ctx);
+        Pointer<CellVariable<NDIM, double> > exact_cc_var = new CellVariable<NDIM, double>("exact_cc", NDIM);
+        const int exact_cc_idx = var_db->registerVariableAndContext(exact_cc_var, ctx);
+#endif
+
+        gridding_algorithm->makeCoarsestLevel(patch_hierarchy, 0.0);
+        const int tag_buffer = std::numeric_limits<int>::max();
+        int level_number = 0;
+        while ((gridding_algorithm->levelCanBeRefined(level_number)))
+        {
+            gridding_algorithm->makeFinerLevel(patch_hierarchy, 0.0, 0.0, tag_buffer);
+            ++level_number;
+        }
+
+        const int finest_level = patch_hierarchy->getFinestLevelNumber();
+        for (int ln = 0; ln <= finest_level; ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            level->allocatePatchData(u_sc_idx, 0.0);
+            level->allocatePatchData(exact_sc_idx, 0.0);
+#ifdef DO_PLOT
+            level->allocatePatchData(u_cc_idx, 0.0);
+            level->allocatePatchData(exact_cc_idx, 0.0);
+#endif
+        }
+
+        HierarchyMathOps hier_math_ops("hier_math_ops", patch_hierarchy);
+
+        {
+            muParserCartGridFunction u_fcn("u", app_initializer->getComponentDatabase("u"), grid_geometry);
+            u_fcn.setDataOnPatchHierarchy(u_sc_idx, u_sc_var, patch_hierarchy, 0.0);
+            u_fcn.setDataOnPatchHierarchy(exact_sc_idx, exact_sc_var, patch_hierarchy, 0.0);
+        }
+        Pointer<VisItDataWriter<NDIM> > visit_writer = app_initializer->getVisItDataWriter();
+
+        // The rest is just book-keeping, this is the actual test:
+        {
+            Pointer<PatchLevel<NDIM> > level_0 = patch_hierarchy->getPatchLevel(0);
+            Pointer<PatchLevel<NDIM> > level_1 = patch_hierarchy->getPatchLevel(1);
+
+            // there should only be one patch on each patch level
+            Pointer<SideData<NDIM, double> > u_fc_0_data = level_0->getPatch(0)->getPatchData(u_sc_idx);
+            Pointer<SideData<NDIM, double> > u_fc_1_data = level_1->getPatch(0)->getPatchData(u_sc_idx);
+            const Box<NDIM> patch_box_0 = level_0->getPatch(0)->getBox();
+            const Box<NDIM> patch_box_1 = level_1->getPatch(0)->getBox();
+
+            // TODO we should probably just read this from the input file.
+            IntVector<NDIM> ratio;
+            for (int d = 0; d < NDIM; ++d)
+            {
+                ratio(d) = 4;
+            }
+            IBTK::CartSideDoubleRT0Coarsen coarsen_op(ratio);
+            coarsen_op.coarsen(*level_0->getPatch(0), *level_1->getPatch(0),
+                               u_sc_idx, u_sc_idx, patch_box_0, ratio);
+            u_fc_1_data->fillAll(0.0);
+        }
+
+#ifdef DO_PLOT
+        visit_writer->registerPlotQuantity(u_cc_var->getName(), "VECTOR", u_cc_idx);
+        hier_math_ops.interp(u_cc_idx, u_cc_var, u_sc_idx, u_sc_var, nullptr, 0.0, false);
```

Comment:
Since everything is happening only on a single patch, you could alternatively use `PatchMathOps` here.